CMakeLists.txt

```diff
@@ -53,6 +53,16 @@ if(UNIX)
   string(APPEND CMAKE_CXX_FLAGS " -Wno-write-strings -Wno-multichar ${BITS} ${EXTRA_CXX_FLAGS}")
   string(APPEND CMAKE_C_FLAGS " ${BITS}")
 
+  if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND BITS MATCHES "-m32")
+    # hacks required to enable 32 bit binaries with a simple -D BITS=-m32 on various 64 bit deb or rpm distros:
+    if(EXISTS "/etc/debian_version")
+      set(CMAKE_LIBRARY_ARCHITECTURE i386-linux-gnu)
+    else()
+      set(CMAKE_SIZEOF_VOID_P 4)
+    endif()
+  endif()
+
+
```

Comment:
Don't add this to CMakeLists.txt. CMAKE_SIZEOF_VOID_P sets by CMake and should'nt be changed by user. Instead of this block just add `"CMAKE_LIBRARY_ARCHITECTURE": "i386-linux-gnu"` into `cacheVariables` of linux-32 preset. Other distros has own approaches to define crosscompile environment.

Reply:
Modifying `CMAKE_SIZEOF_VOID_P` directly in the CMakeLists.txt file isn't a recommended practice because it's a system-defined variable set by CMake based on the architecture it detects. Changing it could lead to unexpected behavior and compatibility issues with other parts of the build system or other projects.

Instead, setting `CMAKE_LIBRARY_ARCHITECTURE` within cache variables of a preset for Linux 32-bit is a cleaner approach. It provides a clearer separation of concerns and ensures that the necessary configurations for cross-compilation are applied in a more standard and maintainable manner.